packages/remix-dev/vite/plugin.ts

```diff
@@ -1075,9 +1075,12 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
               : undefined,
           },
           optimizeDeps: {
-            entries: Object.values(ctx.remixConfig.routes).map((route) =>
-              path.join(ctx.remixConfig.appDirectory, route.file)
-            ),
+            entries: [
+              ctx.entryClientFilePath,
```

Comment:
Following on from this comment: https://github.com/remix-run/remix/pull/9921/files#r1740220074

The client entry is already captured in the Rollup build options, so if that turns out to fix it, it'll cover this case too.